app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -40,7 +41,8 @@ internal fun normalModeAdapterItems(
     showRecentTab: Boolean,
     showRecentSyncedTab: Boolean,
     recentVisits: List<RecentlyVisitedItem>,
-    pocketStories: List<PocketStory>
+    pocketStories: List<PocketStory>,
+    firstFrameDrawn: Boolean = false,
```

Comment:
outside of the updated test, is there ever a scenario where `firstFrameDrawn` is passed-in as `true`?

Reply:
Yeah, below in the `LinearLayoutManager.onLayoutCompleted` call is when we want this flag to be flipped.

Reply:
I saw that's where the flag is being updated, I just wasn't sure if that flag flip then triggered something in `SessionControlView` or if `SessionControlView` was ever re-initialized with `firstFrameDrawn` as true.

Reply:
Ah, I think I understand your comment now. In the `HomeFragment`, we observe the entire `AppState` for changes [here](https://github.com/mozilla-mobile/fenix/blob/f8f8b4f8132366e5b40ef1687d696e4d5a025776/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt#L452-L455) and calls the `sessionControlView.update` method on any state changes which in return re-evaluates the `normalModeAdapterItems` and therefore the `firstFrameDrawn` flag.